Soundex.Tests.cpp

```diff
@@ -1,9 +1,92 @@
 #include <gtest/gtest.h>
 #include "Soundex.h"
 
-TEST(SoudexTestsuite, ReplacesConsonantsWithAppropriateDigits) {
- //AAA
+TEST(SoudexTestsuite, InvalidCharactersSpCharSet1) {
   char soundex[5];
-  generateSoundex("AX", soundex);
- // ASSERT_EQ(soundex,"A200");
+  generateSoundex("A$SAM", soundex);
+  EXPECT_STREQ(soundex,"A250");
```

Comment:
Optimize test code using Value Parametrized Tests in GTest

Reply:
Corrected.